src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollectorHostedService.cs

```diff
@@ -49,20 +55,24 @@ private async Task ExecuteAsync(CancellationToken cancellationToken)
 
             while (!cancellationToken.IsCancellationRequested)
             {
-                _logger.LogDebug("Executing HealthCheck collector HostedService.");
-
-                using (var scope = scopeFactory.CreateScope())
+                // Collect should not be triggered until IServerAddressFeature reports the listening endpoints
+                if (_serverAddressesService.HasAddresses)

```

Comment:
Do not trigger any endpoint collection until the addresses service knows the urls the server is listening on.